pkg/splunk/enterprise/configuration_test.go

```diff
@@ -845,3 +846,71 @@ func TestAreRemoteVolumeKeysChanged(t *testing.T) {
 		t.Errorf("Empty volume should not report a key change")
 	}
 }
+
+func TestAddStorageVolumes(t *testing.T) {
+	labels := make(map[string]string)
+	var replicas int32 = 1
+
+	// Create CR
+	cr := enterprisev1.ClusterMaster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "CM",
+			Namespace: "test",
+		},
+	}
+
+	// create statefulset configuration
+	statefulSet := &appsv1.StatefulSet{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "StatefulSet",
+			APIVersion: "apps/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-statefulset",
+			Namespace: cr.GetNamespace(),
+		},
+		Spec: appsv1.StatefulSetSpec{
+			Replicas: &replicas,
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Image: "test",
+							Name:  "splunk",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Define valid common spec
+	spec := &enterprisev1.CommonSplunkSpec{}
+	err := addStorageVolumes(&cr, spec, statefulSet, labels)
+	if err != nil {
+		t.Errorf("Unable to add storage volumes, error: %s", err.Error())
+	}
+
```

Comment:
Some combinations which need testing
 - When storage class is not explicitly specified, statefulsets are setup with defaults for both etc & var
 - etc with explicit Storage class, var is empty(& hence uses default)
 - var with explicit Storage class, etc is empty(& hence uses default)
 - etc & var with explicit but same Storage classes
 - etc & var with explicit but different Storage classes
 - Changing etc & var's storage class after they have been setup
 - Deleting explicit storage class on etc/var after they have been setup

@pdhanoyasplunk Do you intend to cover these as part of automation tests?

Reply:
I tested most of these scenarios. Dynamic change of storage class name/capacity might not be supported by K8S. The pod might need to be deleted & redeployed for the same.

Reply:
Added UT as well to test the same.